public/wp-content/plugins/zzz-imgproxy/imgproxy.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+Plugin Name: zzz-imgproxy
+Description: Dynamic image resizing
+Author: manGoweb / Mikulas Dite
+Version: 1.0
+Author URI: https://www.mangoweb.cz
+*/
+
+add_action( 'plugins_loaded', 'imgproxy_init' );
+
+require_once ABSPATH . WPINC . '/class-wp-image-editor.php';
+require_once ABSPATH . WPINC . '/class-wp-image-editor-imagick.php';
+
+function imgproxy_init() {
+	if (!defined('IMGPROXY_KEY') || empty(IMGPROXY_KEY)) {
+		throw new \Exception('IMGPROXY_KEY undefined');
+	}
+	if (!defined('IMGPROXY_SALT') || empty(IMGPROXY_SALT)) {
+		throw new \Exception('IMGPROXY_SALT undefined');
+	}
+
+	add_filter('wp_image_editors', 'imgproxy_noop_editor', 50, 1);
+	add_filter('image_downsize', 'imgproxy_image_downsize', 99, 3 );
+}
+
+function imgproxy_image_downsize($param, $id, $size = 'medium') {
+	if ($size === 'full') {
+		return false;
+	}
+
+	// get dimensions for requested size
+	if (is_array($size)) {
+		$width = $size[0];
+		$height = $size[0];
+	} else {
+		$width = get_option("${size}_size_w");
+		$height = get_option("${size}_size_h");
+	}
+
+	// get original url
+	$url = wp_get_attachment_image_url($id, 'full', false);
+	if ($url === false) {
+		return false;
+	}
+
+	return [improxy_url($url, $width, $height), $width, $height, true];
+}
+
+function improxy_url($url, $width, $height) {
+	$keyBin = pack("H*" , IMGPROXY_KEY);
+	if(empty($keyBin)) {
+		die('Key expected to be hex-encoded string');
+	}
+	$saltBin = pack("H*" , IMGPROXY_SALT);
+	if(empty($saltBin)) {
+		die('Salt expected to be hex-encoded string');
+	}
+	$resize = 'fill';
```

Comment:
u `add_image_size` je parametr `$crop`, ten by asi tahle služba měla taky reflektovat.

Reply:
áha super, to jde vytahnout z get options?